### PR TITLE
Fix compile warnings when compiling on latest OSX with clang

### PR DESCRIPTION
**Pull request by normanmaurer**

### CODE REVIEW

**Comment by normanmaurer** [comment]
@jrudolph @nitsanw this makes clang a bit more happy :)

**Comment by nitsanw** [comment]
I'm really sorry, the main blocker for me is that I do not have a MacOS env to test with. I assume you're done your testing and are happy with it. It would be good if we can have the CI working so that at the very least some sanity checks are in place. I'll see if I can resolve that and if the build passes for this on Mac I think it would be safe to merge (I think its safe enough anyhow, but would like to have the pipeline work).

File: src/c/perf-map-agent.c [hunk 1]
```diff
@@ -160,7 +158,7 @@ void write_unfolded_entry(
             //printf("At %d method is %d len %d remaining %d\n", i, info->methods[i], strlen(full_name), sizeof(full_name) - 1 - strlen(full_name));
             sig_string(jvmti, info->methods[i], inlined_name, sizeof(inlined_name));
             strncat(full_name, inlined_name, sizeof(full_name) - 1 - strlen(full_name)); // TODO optimize
-            if (i != 0) strncat(full_name, "->", sizeof(full_name));
+            if (i != 0) strncat(full_name, "->", sizeof(full_name) -1 - strlen(full_name));
```
Comment (jrudolph) [inline src/c/perf-map-agent.c]: Good catch.

Comment (normanmaurer) [reply src/c/perf-map-agent.c]: <3

**Comment by normanmaurer** [comment]
@nitsanw @jrudolph so how can we move this forward ?

**Comment by nitsanw** [comment]
please rebase so CI re-runs your PR, then I'll merge it.
Sorry it took so long for my guilt to overcome me, I'm growing resistant to it...

**Comment by jrudolph** [comment]
I don't the CI ever worked. Or did it?

**Comment by nitsanw** [comment]
just fixed it ;-)

**Comment by normanmaurer** [comment]
@nitsanw woot... build completed :)

**Comment by nitsanw** [comment]
and indeed, no warnings! thanks @normanmaurer

**Comment by normanmaurer** [comment]
@nitsanw <3

**Pull request merged**